src/Symfony/Component/Serializer/Tests/Normalizer/BackedEnumNormalizerTest.php

```diff
@@ -0,0 +1,149 @@
+        $this->assertFalse($this->normalizer->supportsNormalization(UnitEnumDummy::GET));
+        $this->assertFalse($this->normalizer->supportsNormalization(new \stdClass()));
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testNormalize()
+    {
+        $this->assertSame('GET', $this->normalizer->normalize(StringBackedEnumDummy::GET));
+        $this->assertSame(200, $this->normalizer->normalize(IntegerBackedEnumDummy::SUCCESS));
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testNormalizeBadObjectTypeThrowsException()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->normalizer->normalize(new \stdClass());
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testSupportsDenormalization()
+    {
+        $this->assertTrue($this->normalizer->supportsDenormalization(null, StringBackedEnumDummy::class));
+        $this->assertTrue($this->normalizer->supportsDenormalization(null, IntegerBackedEnumDummy::class));
+        $this->assertFalse($this->normalizer->supportsDenormalization(null, UnitEnumDummy::class));
+        $this->assertFalse($this->normalizer->supportsDenormalization(null, \stdClass::class));
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testDenormalize()
+    {
+        $this->assertSame(StringBackedEnumDummy::GET, $this->normalizer->denormalize('GET', StringBackedEnumDummy::class));
+        $this->assertSame(IntegerBackedEnumDummy::SUCCESS, $this->normalizer->denormalize(200, IntegerBackedEnumDummy::class));
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testDenormalizeNullValueThrowsException()
+    {
+        $this->expectException(NotNormalizableValueException::class);
+        $this->normalizer->denormalize(null, StringBackedEnumDummy::class);
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testDenormalizeBooleanValueThrowsException()
+    {
+        $this->expectException(NotNormalizableValueException::class);
+        $this->normalizer->denormalize(true, StringBackedEnumDummy::class);
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testDenormalizeObjectThrowsException()
+    {
+        $this->expectException(NotNormalizableValueException::class);
+        $this->normalizer->denormalize(new \stdClass(), StringBackedEnumDummy::class);
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testDenormalizeBadBackingValueThrowsException()
+    {
+        $this->expectException(NotNormalizableValueException::class);
+        $this->expectExceptionMessage('"POST" is not a valid backing value for enum "'.StringBackedEnumDummy::class.'"');
+        $this->normalizer->denormalize('POST', StringBackedEnumDummy::class);
+    }
+
+    /**
+     * @requires PHP < 8.1
+     */
+    public function testNormalizeShouldThrowExceptionForPHPLowerThan81()
```

Comment:
```suggestion
    public function testNormalizeShouldThrowExceptionForNonEnumObjects()
```

Again, this behavior can be tested on PHP 7 as well.

Reply:
Agreed, fixed this test, `testDenormalizeShouldThrowExceptionForNonEnumObjects` and `testSupportsNormalizationShouldFailOnAnyPHPVersionForNonEnumObjects` as well.